apps/api/src/supplier-invoices/supplier-invoices.controller.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  UsePipes,
+  ValidationPipe
+} from '@nestjs/common';
+import { SupplierInvoicesService } from './supplier-invoices.service';
+import { CreateSupplierInvoiceDto } from './dto/create-supplier-invoice.dto';
+import { UpdateSupplierInvoiceDto } from './dto/update-supplier-invoice.dto';
+
+@Controller('supplier-invoices')
+export class SupplierInvoicesController {
+  constructor(
+    private readonly supplierInvoicesService: SupplierInvoicesService
+  ) {}
+
+  @Post()
+  @UsePipes(new ValidationPipe({ transform: true }))
+  create(@Body() createSupplierInvoiceDto: CreateSupplierInvoiceDto) {
+    return this.supplierInvoicesService.create(createSupplierInvoiceDto);
+  }
+
+  @Get()
+  findAll() {
+    return this.supplierInvoicesService.findAll();
+  }
+
+  @Get(':id')
+  findOne(@Param('id') id: string) {
+    return this.supplierInvoicesService.findOne(+id);
+  }
+
+  @Patch(':id')
+  @UsePipes(new ValidationPipe({ transform: true }))
+  update(
+    @Param('id') id: string,
+    @Body() updateSupplierInvoiceDto: UpdateSupplierInvoiceDto
+  ) {
+    return this.supplierInvoicesService.update(+id, updateSupplierInvoiceDto);
+  }
```

Comment:
I think the Patch should come first. Can you clarify why this should be reversed?

Reply:
I'm confused because in a different comment you recommended that the route decorator should be the outermost decorator. If decorators in TypeScript are applied in a bottom-up manner, then isn't the order that I have it correct if we want validation to occur before the route handling logic?

You can also see that in the Nest.js documentation: https://docs.nestjs.com/techniques/validation, the order is the way I have it applied.